src/reducers/index.js

```diff
@@ -287,19 +291,26 @@ const getTxHistoryFromWSTx = (tx, tokenUid, tokenTxBalance) => {
 
 /**
  * Got wallet history. Update wallet data on redux
+ * @param {string[]} action.payload.tokens
+ * @param {string} action.payload.currentAddress
+ * @param {{uid:string, name:string, symbol:string}[]} action.payload.registeredTokens
  */
 const onLoadWalletSuccess = (state, action) => {
   // Update the version of the wallet that the data was loaded
   LOCAL_STORE.setWalletVersion(VERSION);
   const { tokens, registeredTokens, currentAddress } = action.payload;
-  const allTokens = new Set(tokens);
 
+  // Convert the tokens list to a new object to ensure there are no duplicated tokens
+  const allTokens = {};
+  for (const uid of tokens) {
+    allTokens[uid] = uid;
+  }
```

Comment:
I don't like this in the reducer, I think that the `tokens` should already be in the converted format when they reach this point

Can you move it to the saga?

It would also be nice to move this `LOCAL_STORE.setWalletVersion(VERSION);`, which is a side-effect to the saga, it shouldn't be here

Reply:
I agree the reducer is not the place to transform an array into an object map, so I extracted the logic for the deduplication of allTokens on https://github.com/HathorNetwork/hathor-wallet/pull/587/commits/6b87012bfa353140c7a864a832f81fdc097a799d .

However this comes with a tradeoff: It is now the responsibility of the action caller to pass a new object as a parameter.

As it was mentioned on [a comment below](https://github.com/HathorNetwork/hathor-wallet/pull/587#discussion_r1576573372), if the object is not completely new ( for example, the caller function mutates the object and passes it again as a parameter ), weird behaviors may happen on the application because of a lack of re-renders.

This should be improved as soon as we have the full Redux tools available, after the major upgrade of the Redux dependency.

The `LOCAL_STORE.setWalletVersion` alteration should be made in a future PR, as it is beyond the scope of this PR.

Reply:
Opened issue #588 to track reducers with side effects such as this one.